src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleObject.cs

```diff
@@ -16,11 +16,19 @@ private static void HandleStartObject(JsonSerializerOptions options, ref ReadSta
             if (state.Current.IsProcessingEnumerable())
             {
                 // A nested object within an enumerable.
+
+                if (!state.Current.CollectionPropertyInitialized)
+                {
+                    // We have bad JSON: enumerable element appeared without preceding StartArray token.
+                    Debug.Assert(state.Current.JsonPropertyInfo != null);
```

Comment:
Is there any value in this `Debug.Assert` ? It will dereference null on the next line 10ns later and break into the debugger/crash anyway. Generally `Debug.Assert` is useful when it catches an issue that might not otherwise cause bad consequences tli later.